examples/showcase_1kt.rs

```diff
@@ -0,0 +1,240 @@
+const N_MOTES: usize = 2;
+
+/// Interval between frame transmissions.
+const N_SECS: u32 = 3;
+const SLEEP_TIME: Milliseconds = Milliseconds(1000 * N_SECS);
+
+fn get_cherry_id() -> Option<&'static str> {
+    option_env!("HENI_DEVICE")
+}
+
+struct TemperatureDisplay(i32);
+
+impl TemperatureDisplay {
+    const MAX_DISPLAYED_BYTES_LEN: usize =
+        1 /* sign */ +
+        4 /* +-1000*C is for sure a bound for measured temperature */ +
+        1 /* decimal dot */ +
+        2 /* decimal points */ +
+        2 /* *C */;
+}
+
+impl Display for TemperatureDisplay {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "{}{}.{}*C",
+            if self.0 < 0 { "-" } else { "" },
+            i32::abs(self.0) / 100,
+            i32::abs(self.0) % 100
+        )
+    }
+}
+
+const U32_MAX_DISPLAY_LEN: usize = 10;
+
+macro_rules! msg_template {
+    () => {
+        "Timestamp {}, CherryMote {}, measurement no. {}, temperature {}."
+    };
+}
+const MSG_TEMPLATE_REAL_LEN: usize = msg_template!().len() - 4 /* number of substitutions */ * 2 /* pair of braces */;
+
+const MSG_BUF_LEN: usize = MSG_TEMPLATE_REAL_LEN
+    + U32_MAX_DISPLAY_LEN /* timestamp */
+    + U32_MAX_DISPLAY_LEN /* CherryMote id */
+    + TemperatureDisplay::MAX_DISPLAYED_BYTES_LEN;
+
+struct MsgBuf<const N: usize> {
+    buf: [u8; N],
+    offset: usize,
+}
+impl<const N: usize> MsgBuf<N> {
+    fn new() -> Self {
+        Self {
+            buf: [0_u8; N],
+            offset: 0,
+        }
+    }
+
+    fn fill(
+        &mut self,
+        timestamp: u32,
+        sequence_no: usize,
+        temperature_centigrades_celcius: i32,
+    ) -> usize {
+        let temperature_displayer = TemperatureDisplay(temperature_centigrades_celcius);
+
+        // Each filling start overrides the buffer.
+        self.offset = 0;
+
+        struct Writer<'a> {
+            buf: &'a mut [u8],
+            offset: &'a mut usize,
+        }
+        impl core::fmt::Write for Writer<'_> {
+            fn write_str(&mut self, s: &str) -> core::fmt::Result {
+                let prefix = &mut self.buf[*self.offset..*self.offset + s.len()];
+                prefix.copy_from_slice(s.as_bytes());
+                *self.offset += s.len();
+                Ok(())
+            }
+        }
+
+        let mut w = Writer {
+            buf: &mut self.buf,
+            offset: &mut self.offset,
+        };
+
+        writeln!(
+            w,
+            msg_template!(),
+            timestamp,
+            get_cherry_id().unwrap_or("<unknown id>"),
+            sequence_no,
+            temperature_displayer
+        )
+        .unwrap();
+
+        self.offset
+    }
+
+    fn inner(&self) -> &[u8] {
+        &self.buf
+    }
+}
+
+fn main() {
+    // Configure the radio
+    let pan: u16 = 0xcafe;
+    let addr_short: u16 = 0xdead;
+    let addr_long: u64 = 0xdead_bead;
+    let tx_power: i8 = 5;
+    let channel: u8 = 11;
```

Comment:
Z tego zrobiłbym const'y na początku pliku lub nawet w oddzielnym conf.rs, bo to wygląda na coś, co można chcieć przestawiać przy deploymencie i chciałoby się uniknąć szukania tego gdzieś w głębi pliku.